spacy/pipeline/span_ruler.py

```diff
@@ -0,0 +1,572 @@
+    spans_filter: Optional[Callable[[Iterable[Span], Iterable[Span]], Iterable[Span]]],
+    annotate_ents: bool,
+    ents_filter: Callable[[Iterable[Span], Iterable[Span]], Iterable[Span]],
+    phrase_matcher_attr: Optional[Union[int, str]],
+    validate: bool,
+    overwrite: bool,
+    sort_key: Optional[Callable[[Span], Any]],
+    sort_reverse: bool,
+    scorer: Optional[Callable],
+):
+    return SpanRuler(
+        nlp,
+        name,
+        spans_key=spans_key,
+        spans_filter=spans_filter,
+        annotate_ents=annotate_ents,
+        ents_filter=ents_filter,
+        phrase_matcher_attr=phrase_matcher_attr,
+        validate=validate,
+        overwrite=overwrite,
+        sort_key=sort_key,
+        sort_reverse=sort_reverse,
+        scorer=scorer,
+    )
+
+
+def overwrite_overlapping_ents_filter(
+    entities: Iterable[Span], spans: Iterable[Span]
+) -> List[Span]:
+    """Merge entities and spans into one list without overlaps by allowing
+    spans to overwrite any entities that they overlap with. Priority is given
+    to the earlier spans in the provided spans. Intended to replicate the
+    overwrite_ents=True behavior from the EntityRuler.
+
+    entities (Iterable[Span]): The entities, already filtered for overlaps.
+    spans (Iterable[Span]): The spans to merge, may contain overlaps.
+    RETURNS (List[Span]): Filtered list of non-overlapping spans.
+    """
+    entities = list(entities)
+    new_entities = []
+    seen_tokens: Set[int] = set()
+    for span in spans:
+        start = span.start
+        end = span.end
+        # check for end - 1 here because boundaries are inclusive
+        if start not in seen_tokens and end - 1 not in seen_tokens:
+            new_entities.append(span)
+            entities = [e for e in entities if not (e.start < end and e.end > start)]
+            seen_tokens.update(range(start, end))
+    return entities + new_entities
+
+
+@registry.misc("spacy.overwrite_overlapping_ents_filter.v1")
+def make_overwrite_overlapping_ents_filter():
+    return overwrite_overlapping_ents_filter
+
+
+def preserve_existing_ents_filter(
+    entities: Iterable[Span], spans: Iterable[Span]
+) -> List[Span]:
+    """Merge entities and spans into one list without overlaps by prioritizing
+    existing entities. Intended to replicate the overwrite_ents=False behavior
+    from the EntityRuler.
+
+    entities (Iterable[Span]): The entities, already filtered for overlaps.
+    spans (Iterable[Span]): The spans to merge, may contain overlaps.
+    RETURNS (List[Span]): Filtered list of non-overlapping spans.
+    """
+    entities = list(entities)
+    new_entities = []
+    seen_tokens: Set[int] = set()
+    seen_tokens.update(*(range(ent.start, ent.end) for ent in entities))
+    for span in spans:
+        start = span.start
+        end = span.end
+        # check for end - 1 here because boundaries are inclusive
+        if start not in seen_tokens and end - 1 not in seen_tokens:
```

Comment:
Same comment about having to check all tokens instead of just `start` and `end-1`